src/charm.py

```diff
@@ -946,58 +947,86 @@ def _plugin_akismet_reconciliation(self):
+
+        Args:
+            php_code: PHP code to be executed.
+
+        Returns:
+            An instance of :attr:`charm.WordpressCharm._ExecResult`.
+        """
+        return self._wrapped_run_wp_cli(["wp", "eval", php_code])
+
     @staticmethod
     def _encode_openid_team_map(team_map):
         """Convert wp_plugin_openid_team_map setting to WordPress openid_teams_trust_list option
+        in PHP code.
 
         example input: site-sysadmins=administrator,site-editors=editor,site-executives=editor
 
         Args:
             team_map (str): team definition.
 
         Returns:
-            A serialized PHP array, as a Python string.
+            A PHP array, as a Python string.
         """
-        team_map_lines = []
-        i = 0
-        team_map_lines.append("a:{}:{{".format(len(team_map.split(","))))
-        for mapping in team_map.split(","):
-            i = i + 1
-            team, role = mapping.split("=", 2)
-            team_map_lines.append("i:{};".format(i))
-            team_map_lines.append('O:8:"stdClass":4:{')
-            team_map_lines.append('s:2:"id";')
-            team_map_lines.append("i:{};".format(i))
-            team_map_lines.append('s:4:"team";')
-            team_map_lines.append('s:{}:"{}";'.format(len(team), team))
-            team_map_lines.append('s:4:"role";')
-            team_map_lines.append('s:{}:"{}";'.format(len(role), role))
-            team_map_lines.append('s:6:"server";')
-            team_map_lines.append('s:1:"0";')
-            team_map_lines.append("}")
-        team_map_lines.append("}")
-
-        return "".join(team_map_lines)
+        array_items = []
+        for idx, mapping in enumerate(team_map.split(","), start=1):
+            launchpad_role, wordpress_role = mapping.split("=")
+            launchpad_role = launchpad_role.strip()
+            wordpress_role = wordpress_role.strip()
+            array_items.append(
+                f"{idx} => (object) array ("
+                f"'id'=>{idx},"
+                f"'team'=>'{launchpad_role}',"
+                f"'role'=>'{wordpress_role}',"
+                f"'server' => '0',),"
+            )
+        return f"array({''.join(array_items)})"
 
     def _plugin_openid_reconciliation(self):
         """Reconciliation process for the openid plugin"""
         openid_team_map = self.model.config["wp_plugin_openid_team_map"].strip()
+        result = None
+
+        def check_result():
+            if not result.success:
+                raise exceptions.WordPressBlockedStatusException(
+                    f"Unable to config openid plugin, {result.message}"
+                )
+
         if not openid_team_map:
+            result = self._wp_option_update("users_can_register", "0")
+            check_result()
             result = self._deactivate_plugin(
-                "openid", ["openid_required_for_registration", "openid_teams_trust_list"]
+                'wordpress-teams-integration', ["openid_teams_trust_list"]
```

Comment:
Does this pass linting? I'd expect double quotes here

Reply:
It was causing by the current black configuration. Fixed, thanks!